src/components/topRail/TopRail.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <div class="top-rail-container">
+    <ValidateIssue
+      v-show="isOpenIssue"
+      :number-of-errors="numberOfErrors"
+      @openPanel="handleOpenPanel"
+    />
+
+    <ValidateButton @openIssue="handleOpenIssue" />
+
+    <ValidatePanel
+      v-show="isOpenIssue && isOpenPanel"
+      :error-list="errorList"
+      :warnings="warnings"
+    />
+  </div>
+</template>
+
+<script>
+import store from '@/store';
+import { ValidateButton, ValidateIssue, ValidatePanel } from '@/components/topRail/validateControl';
+
+export default {
+  components: {
+    ValidateButton,
+    ValidateIssue,
+    ValidatePanel,
+  },
+  props: ['validationErrors', 'warnings'],
+  data() {
+    return {
+      isOpenIssue: false,
+      isOpenPanel: false,
+    };
+  },
+  computed: {
+    errorList() {
+      // Get a formatted error list to show in the issue panel
+      return Object.entries(this.validationErrors)
+        .flatMap(([errorKey, errors]) => (
+          errors.flatMap(error => ({
+            ...error,
+            errorKey,
+            ...{ 'errorId': `${error.id}_${error.message.split(' ').join('_')}` },
+          }))
+        ));
+    },
+    numberOfErrors() {
+      // Get the number of errors
+      return this.errorList.length + this.warnings.length;
+    },
+  },
+  watch: {
+    numberOfErrors(newValue) {
+      // Checks the number of errors, if it is "0" hides the panel errors
+      if (newValue === 0) {
+        this.isOpenPanel = false;
+      }
+    },
+  },
+  methods: {
+    handleOpenIssue(value) {
+      // Show/hide the issue button
+      this.isOpenIssue = value;
+      // Set the auto-validate value store
+      store.commit('setAutoValidate', this.isOpenIssue);
+    },
+    handleOpenPanel(value) {
```

Comment:
add documentations for the methods use JDoc

Reply:
It will be added